### DISCUSSION TITLE
What are the differences between chat vs cai-chat mode? How does character extension work with logs?

### DISCUSSION THREAD

**Discussion by tensiondriven** [oobabooga/text-generation-webui#700] [Q&A]
Hola - I've been using text-generation-webui with cai-chat mode, and loving it; I'm curious, what are the differences between cai-chat and chat modes?  Specifically, I'm wondering about how the system behaves behind the scenes.

- Are the log file jsons in ./logs saved differently when in chat vs cai-chat?  Or is this a concern of the "characters" extension?
- Is the model prompted differently based on chat vs cai-chat?
- Do the character loading mechanisms work differently?

If we can get some of this documented in a discussion, I'd be happy to update the documentation to make it clearer for others (or, perhaps this discussion will suffice)

Thanks!

**Comment by oobabooga** [comment]
`cai-chat` uses a custom HTML generator that I wrote from scratch. `chat` uses the gradio chat widget

**Comment by bmoconno** [comment]
To expand on what has already been said, just to add context for people looking at this thread in the future, both `chat` and `cai-chat` use the same `chatbot_wrapper` method in `chat.py` for text generation, but when you are using `cai-chat` it calls that method from it's own `cai_chatbot_wrapper` that additionally generates the HTML for the `cai-chat` from the output of the `chatbot_wrapper` method.

**Comment by PladsElsker** [comment]
Interesting, is one of them better than the other for certain tasks? What are the pros and cons of each mode?
